sci-libs/opencascade/opencascade-7.5.2.ebuild

```diff
@@ -0,0 +1,175 @@
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtquickcontrols2:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtxml:5
+	)
+	ffmpeg? ( media-video/ffmpeg )
+	freeimage? ( media-libs/freeimage )
+	json? ( dev-libs/rapidjson )
+	tbb? ( dev-cpp/tbb )
+	vtk? ( sci-libs/vtk[rendering] )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	doc? ( app-doc/doxygen )
+	examples? ( dev-qt/linguist-tools:5 )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-7.5.1-0001-allow-default-search-path-for-Qt5.patch
+	"${FILESDIR}"/${PN}-7.5.1-0002-remove-unnecessary-Qt5-check.patch
+	"${FILESDIR}"/${PN}-7.5.1-0003-add-Gentoo-configuration-type.patch
+	"${FILESDIR}"/${PN}-7.5.1-0004-fix-installation-of-cmake-config-files.patch
+	"${FILESDIR}"/${PN}-7.5.1-0005-fix-write-permissions-on-scripts.patch
+	"${FILESDIR}"/${PN}-7.5.1-0006-fix-creation-of-custom.sh-script.patch
+	"${FILESDIR}"/${PN}-7.5.1-fix-AllValues-name-collision-with-vtk-9.0.patch
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use debug; then
+		append-cppflags -DDEBUG
+		append-flags -g
+	fi
+
+	sed -e 's/\/lib\$/\/'$(get_libdir)'\$/' \
+		-i adm/templates/OpenCASCADEConfig.cmake.in || die
+
+	# There is an OCCT_UPDATE_TARGET_FILE cmake macro that fails due to some
+	# assumptions it makes about installation paths. Rather than fixing it, just
+	# get rid of the mechanism altogether - its purpose is to allow a
+	# side-by-side installation of release and debug libraries.
+	sed -e 's|\\${OCCT_INSTALL_BIN_LETTER}||' \
+		-i "adm/cmake/occt_toolkit.cmake" || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_DOC_Overview=$(usex doc)
+		-DBUILD_Inspector=$(usex examples)
+		-DBUILD_WITH_DEBUG=$(usex debug)
+		-DCMAKE_INSTALL_PREFIX="/usr"
+		-DINSTALL_DIR_BIN="$(get_libdir)/${P}/bin"
+		-DINSTALL_DIR_CMAKE="$(get_libdir)/cmake/${P}"
+		-DINSTALL_DIR_DOC="share/doc/${PF}"
+		-DINSTALL_DIR_LIB="$(get_libdir)/${P}"
+		-DINSTALL_DIR_SCRIPT="$(get_libdir)/${P}/bin"
+		-DINSTALL_DIR_WITH_VERSION=ON
+		-DINSTALL_SAMPLES=$(usex examples)
+		-DINSTALL_TEST_CASES=NO
+		-DUSE_D3D=NO
+		-DUSE_FFMPEG=$(usex ffmpeg)
+		-DUSE_FREEIMAGE=$(usex freeimage)
+		-DUSE_FREETYPE=ON
+		-DUSE_GLES2=$(usex gles2)
+		-DUSE_RAPIDJSON=$(usex json)
+		-DUSE_TBB=$(usex tbb)
+		-DUSE_VTK=$(usex vtk)
+	)
+
+	use doc && mycmakeargs+=( -DINSTALL_DOC_Overview=ON )
+
+	if use examples; then
+		mycmakeargs+=(
+			-D3RDPARTY_QT_DIR="${ESYSROOT}"/usr
+			-DBUILD_SAMPLES_QT=ON
+		)
+	fi
+
+	if use vtk; then
+		if has_version ">=sci-libs/vtk-9.0.0"; then
+			mycmakeargs+=(
+				-D3RDPARTY_VTK_DIR="${ESYSROOT}"/usr
+				-D3RDPARTY_VTK_INCLUDE_DIR="${ESYSROOT}"/usr/include/vtk-9.0
+				-D3RDPARTY_VTK_LIBRARY_DIR="${ESYSROOT}"/usr/$(get_libdir)
+			)
+		fi
+	fi
+
+	cmake_src_configure
+
+	prepare_env_file() {
+		# prepare /etc/env.d file
+		sed -e 's|VAR_CASROOT|'${ESYSROOT}'/usr|g' < "${FILESDIR}/${PN}-${PV_MAJ}.env.in" >> "${T}/${PV_MAJ}" || die
+		sed -e 's|lib/|'$(get_libdir)'/|g' -i "${T}/${PV_MAJ}" || die
+		sed -e 's|VAR_PV|'${PV}'|g' -i "${T}/${PV_MAJ}" || die
```

Comment:
1. I'd count this as personal style preference and not necessarily something relevant for a review. Or do I miss a point in why this *should* be combined?
2. good catch
3. I had cases in the past, where bash expansions didn't work as expected, most notably `$(get_libdir)` not getting expanded at all, so I sticked to quoting most variables, if I'm not sure it get's expanded.
3.5 I don't understand. The env file is quoted everywhere in the sed commands
4. Might be, I got the working of the `g` flag wrong. I supposed sed will change only the first match without it. A quick test yet revealed, that sed changes all matches even without it applied. I'm gonna fix this.

I kindly ask you not to provide suggestions for changes, as long as I do not ask your for one. For me this is counter-productive and significantly decreases my learning curve. I'm usually well prepared to do my own research and try out different possible solutions for myself. I'm going to ask for help, if I'm stuck. In the above case, it can also give the impression that you might want to force your personal style preference onto others (c.f. bullet point 1), which I see as bad thing (I'm not saying you want to do this, just that the impression can be given to others).

Reply:
1. Well, "should" was a strong word, but it's extra calls, writes, filenames, and `|| die` for nothing, and combining doesn't really make it less readable (still on 1 line each) so I don't really get why you wouldn't.

3. point is just that it as it is, it'll fail if ESYSROOT contains spaces

3.5. the env file contains `CASROOT=VAR_CASROOT` rather than `CASROOT="VAR_CASTROOT"` you replace VAR_CASROOT, which can result in `CASROOT=/some path with spaces/usr`, sourcing this will fail at runtime when running env-update

I didn't mean /have/ to use as-is, Suggestion is just to help get the point across, but I'll keep that in mind.

Reply:
1. I didn't said I wouldn't consider it. I just won't do it, because someone is saying I should, without giving any reasoning. The reason you've given now are good arguments for combining the statements.
3. good example, which will also fail if you use single quotes instead of double quotes, e.g. `ESYSROOT="/some path/usr"; sed -e 's|a|${ESYSROOT}|' < file` will not expand ESYSROOT, only if you use double quotes, the variable will be expanded. Most example scripts in the sed info file use single quotes, so I must assume this is somehow preferred. And it has the plus, that you can use double quotes within the expression without the need of an extra quote for them.

I didn't get it the way that I *have* to use your suggestion. If I don't agree, I can simply ignore it. My comment was just a suggestion from my POV on possible downsides of using suggestions.

Reply:
I'm confused, why would you use single quotes when there is a variable to expand? Single quotes are there to prevent things like expansion in the first place.

sed often use single quotes because they use all sort of special characters like `$` as end-of-line character, and single quote is a safety against that. Idea is to use what is more suitable for what you're doing with the sed.

Reply:
Was looking at your newer seds, there was no need to change them all. It's only when you want to expand variables.

`sed -e "s|^#MMGT_OPT=0$|MMGT_OPT=2|"` doesn't have variables to expand, so using single quotes is fine.

Reason people would want to use single quotes here is, for example, say you had been using `s_=0$_=2_` instead of `s|=0$|=2|` $_ is a valid variable in bash and would get expanded to something else. Thankfully `$|` isn't, so this isn't broken and can be left like that if you prefer.

Just saying, use of both kind of quotes here isn't about style but serves a purpose.

Reply:
The decision to change from single to double quotes was a question of style. I wanted to use one style as far as possible. Definitely need to get into sed some more. I'm just a basic sed user, most of it's functionality is still a miracle to me :)

> I'm confused, why would you use single quotes when there is a variable to expand? Single quotes are there to prevent things like expansion in the first place.

I wasn't aware of this. I've always seen this as some kind of style with respect to sed.